code/modules/ai/ai_behaviors/ai_behavior.dm

```diff
@@ -103,6 +105,10 @@ Registers signals, handles the pathfinding element addition/removal alongside ma
 	current_node.set_weight(identifier, NODE_LAST_VISITED, world.time)
 	change_action(MOVING_TO_NODE, current_node)
 
+///Signal handler when the ai is blocked by an obstacle
+/datum/ai_behavior/proc/deal_with_obstacle(datum/source, direction)
+	SIGNAL_HANDLER
```

Comment:
this adds pointless overhead for children that dont have this implemented, it should only register if this actually has something